test.sh

```diff
@@ -5,11 +5,70 @@ if [ -f /.dockerenv ]; then
     export PATH=${PATH}:/bitcoin/bin:/liquid/bin
 fi
 
-bitcoind -server=1 -regtest -daemon
+DATADIR=${DATADIR:-~/tmp/test_gdk_rpc}
+EXEC_NODE=${EXEC_NODE:-bitcoind}
+EXEC_CLI=${EXEC_CLI:-bitcoin-cli}
 
-bitcoin-cli -regtest -rpcwait generatetoaddress 200 $(bitcoin-cli -regtest -rpcwait getnewaddress)
+TEST_MODE=${TEST_MODE:-bitcoinregtest}
 
-BITCOIND_DIR=~/.bitcoin/regtest \
-cargo test --features stderr_logger --all  -- --test-threads=1
 
-bitcoin-cli -regtest stop
+echo "Killing existing nodes..."
+killall -9 $EXEC_NODE || true
```

Comment:
not a fan of this, it closed the mainnet node on my machine, why change the `bitcoin-cli -regtest stop` which target only regtest node?

Reply:
Hmm, I remember changing this for an edge case, probably should have changed this back. I think somehow nodes got unresponsive during startup or wallet loading time or so. Will revert and do manual kill in the occasional issue.